dissect/target/filesystem.py

```diff
@@ -357,18 +364,20 @@ def readlink_ext(self, path: str) -> FilesystemEntry:
         """
         return self.get(path).readlink_ext()
 
-    def stat(self, path: str) -> fsutil.stat_result:
-        """Determine the stat information of a ``path`` on the filesystem, resolving any symlinks.
+    def stat(self, path: str, follow_symlinks: bool = True) -> fsutil.stat_result:
+        """Determine the stat information of a ``path`` on the filesystem.
 
-        If the path is a symlink, it gets resolved, attempting to stat the path where to points to.
+        If ``path`` is a symlink and ``follow_symlinks`` is ``True``, it gets resolved, attempting to stat the
+        path where it points to.
 
         Args:
             path: The filesystem path we want the stat information from.
+            follow_symlinks: Whether to resolve the path if it is a symbolic link.
 
         Returns:
             The stat information of the given path.
         """
-        return self.get(path).stat()
+        return self.get(path).stat(follow_symlinks=follow_symlinks)
```

Comment:
Could this one not be optimized to directly call `self.lstat()` is `follow_symlinks == False`?

Reply:
Ah, I wasn't aware that this was due to `os.walk`. It hasn't been changed in 3.11 or 3.12, so I'll remove the commit that changes this.